### PR TITLE
Pure autonomous mission controller w/o bootutility

### LINKED ISSUE #619: [PTest] - Pure Autonomous Mission Controller [Framework Only]
# Why
We need a PTest case (or some modification of one), to eventually evolve into or just be, the system that will run the mission of Satellites once they are launched.

# What we have right now
Pending Andreea's PR, we will have an AMC, but it currently also is responsible for booting HITL/HOOTL instances.

# What we need to do
- We need to divorce the current AMC of actually "starting" any HITL/HOOTL instances or at least a way to configure a PTest case to do so
- Specifically, AMC should be configured to connect to exactly two RadioSessions, as this is the flight-like condition

# AMC's specific features to keep in mind (can be deferred to future tickets):
- AMC must uplink data about one satellite to the other, and vice versa so that they can perform a successful rendezvous
- AMC's automatic uplinks must be able to "mesh" nicely with manual commands/uplinks
- AMC must make intelligent decisions upon when to set satellites into specific modes, such as setting spacecraft roles
- Further documentation is found here: https://github.com/pathfinder-for-autonomous-navigation/FlightSoftware/issues/476

# What's after this?
Once the AMC framework is created, the we can then start on giving the AMC its actual brains and features outlined here:
https://github.com/pathfinder-for-autonomous-navigation/FlightSoftware/issues/476

### PR DESCRIPTION
**Pull request by afoarce**
# Pure AMC w/o bootutility

Fixes #619 

### Summary of changes
- Added checks for if config json has any devices before initializes leader/follower for mission cases to avoid booting up hitl/hootl instances of the satellites
- AMC assumes config file has two radios, for testing can be the same radio
- Original Autonomous Mission Manager(autonomous_mission_manager_case.py) copied/modified into autonomous_mission_manager_pure_radio.py

Future work includes testing and fixing the RadioSession interactions, as described in #622 and #629 

### Testing
Changes tested in HOOTL but as noted above needs to be tested further.

### CODE REVIEW

File: ptest/cases/base.py [hunk 1]
```diff
@@ -426,30 +428,32 @@ def fast_boot_follower(self):
...
-        self.boot_util_follower = BootUtil(
-            self.flight_controller_follower, self.logger, self.initial_state_follower, 
-            self.fast_boot_follower, self.one_day_ccno_follower, self.suppress_faults)
-        self.boot_util_leader.setup_boot()
-        self.boot_util_follower.setup_boot()
-        self.setup_post_bootsetup_leader()
-        self.setup_post_bootsetup_follower()
+        if self.devices != None:
+            self.setup_pre_bootsetup_leader()
+            self.setup_pre_bootsetup_follower()
```
Comment (shihaocao) [inline ptest/cases/base.py:433]: expose this out of the != None

Comment (shihaocao) [reply ptest/cases/base.py:433]: nvm

File: ptest/cases/autonomous_mission_manager_pure_radio.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    def sim_model(self):
+        return DualAttitudeOrbitGnc
+
+    @property
+    def sim_mapping(self):
+        return "ci_mapping.json"
+        
+    @property
+    def sim_duration(self):
+        return float("inf")
```
Comment (shihaocao) [inline ptest/cases/autonomous_mission_manager_pure_radio.py:71]: still running psim, we should take this out

Comment (shihaocao) [reply ptest/cases/autonomous_mission_manager_pure_radio.py:71]: set float("inf") -> 0

File: ptest/cases/autonomous_mission_manager_pure_radio.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    def sim_model(self):
+        return DualAttitudeOrbitGnc
+
+    @property
+    def sim_mapping(self):
+        return "ci_mapping.json"
+        
+    @property
+    def sim_duration(self):
+        return 0
```
Comment (kylekrol) [inline ptest/cases/autonomous_mission_manager_pure_radio.py:71]: Just curious here why we are providing simulation information? Shouldn't we not be using a simulation at all?

Comment (kylekrol) [reply ptest/cases/autonomous_mission_manager_pure_radio.py:71]: Discussed on Discord, these fields need to remain to avoid a crash.

**Review by kylekrol — COMMENTED** [review]
Seems pretty good to me. Although I'll defer to @shihaocao for a more in depth review - I'm still iffy on `ptest` related things.

**Review by shihaocao — APPROVED** [review]

**Comment by shihaocao** [comment]
Looks good. Let's demo this asap

**Pull request merged**